### ISSUE TITLE
Does crawler.discoverResources exist?

### ISSUE THREAD

**Issue by ajkochanowicz** [cgiffard/node-simplecrawler#106]
I see an option to turn off crawler.discoverResources to prevent simplecrawler from discovering resources on a page. However, not only does it not work in my implementation, I don't see any check for this value in the source code. It looks like SC just runs this function regardless of user settings. Am I wrong?

**Comment by talss89** [comment]
We ran into this problem too. I've created a fork which fixes this, and also the null header bug that prevents SC from running on Node 0.12.0. 

I've written some tests, but this is more of a quick fix to hit a deadline rather than a PR candidate. A PR has already been raised for the header bug #107 

On my fork, set crawler.useDefaultDiscovery to a function to override the default regex discovery. The callback should look like:

```
function(resourceText, queueItem) {
    return ["array", "of", "urls"];
}
```

Perhaps we could discuss this here and create a long-term solution from my hacky code?
